src/mount.rs

```diff
@@ -0,0 +1,655 @@
+    DirectoryEntry, DirectoryInfo, FileInfo, Notification, ProjectedFileSystem,
+    ProjectedFileSystemSource,
+};
+
+use crate::{
+    counters::Counter,
+    monitor::{
+        task::{Task, TaskList},
+        Monitor, Problem,
+    },
+    Apath, Archive, BandId, BandSelectionPolicy, IndexEntry, IndexRead, Kind, Result, StoredTree,
+};
+
+macro_rules! static_dir {
+    ($name:expr) => {
+        DirectoryInfo {
+            directory_name: ($name).to_string(),
+            directory_attributes: DIRECTORY_ATTRIBUTES,
+
+            ..Default::default()
+        }
+        .into()
+    };
+}
+
+struct VoidMonitor;
+impl Monitor for VoidMonitor {
+    fn count(&self, _counter: Counter, _increment: usize) {}
+
+    fn set_counter(&self, _counter: Counter, _value: usize) {}
+
+    fn problem(&self, _problem: Problem) {}
+
+    fn start_task(&self, name: String) -> Task {
+        /*
+         * All data related to the target task will be dropped
+         * as soon the callee drops the task.
+         */
+        let mut list = TaskList::default();
+        list.start_task(name)
+    }
+}
+
+#[derive(Debug)]
+struct HunkMetaInfo {
+    index: u32,
+
+    start_path: Apath,
+    end_path: Apath,
+}
+
+struct HunkHelper {
+    hunks: Vec<HunkMetaInfo>,
+}
+
+impl HunkHelper {
+    pub fn from_index(index: &IndexRead) -> Result<Self> {
+        let mut hunk_info = index
+            .hunks_available()?
+            .into_par_iter()
+            .map(move |hunk_index| {
+                let mut index = index.duplicate();
+                let entries = index.read_hunk(hunk_index)?;
+                let meta_info = if let Some(entries) = entries {
+                    if let (Some(first), Some(last)) = (entries.first(), entries.last()) {
+                        Some(HunkMetaInfo {
+                            index: hunk_index,
+
+                            start_path: first.apath.clone(),
+                            end_path: last.apath.clone(),
+                        })
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                };
+
+                Ok(meta_info)
+            })
+            .map(Result::ok)
+            .flatten()
+            .filter_map(|entry| entry)
+            .collect::<Vec<_>>();
+
+        /* After parallel execution bring all hunks back into order */
+        hunk_info.sort_by_key(|info| info.index);
+        Ok(Self { hunks: hunk_info })
+    }
+
+    pub fn find_hunk_index_for_file(&self, path: &Apath) -> Option<usize> {
+        let hunk_index = self.hunks.binary_search_by(|entry| {
+            match (entry.start_path.cmp(path), entry.end_path.cmp(path)) {
+                (Ordering::Less, Ordering::Less) => Ordering::Less,
+                (Ordering::Greater, Ordering::Greater) => Ordering::Greater,
+                _ => Ordering::Equal,
+            }
+        });
+
+        let hunk_index = match hunk_index {
```

Comment:
maybe this could do with a comment about the interpretation of Err from binary_search

Reply:
That's actually a relict where I only compared the start path and not the start and end path, therefore never encountering an actual match.